### ISSUE TITLE
Unused parameter in array destructing

### ISSUE THREAD

**Issue by fxOne** [SonarSource/SonarJS#1019]
**I want to report a bug.**

SonarJS version: 4.1 (build 6085)
SonarQube version: Version 7.2.1 (build 14109)

**Rule key:** javascript:UnusedVariable and javascript:S1854

**Reproducer** 
```
const myMap = new Map();
for(const [key, value] of myMap) { 
  fn(value);
}
```
The code triggers the following issues:
Remove the declaration of the unused 'key' variable. (javascript:UnusedVariable)
Remove this useless assignment to local variable "key" (avascript:S1854) 

**Expected behavior**
There shouldn't be an issue as I use the 2nd parameter.

**Comment by vilchik-elena** [comment]
@fxOne can't you omit unused elements of array destructuring?
```js
const myMap = new Map();
for(const [, value] of myMap) { 
  fn(value);
}
```

**Comment by fxOne** [comment]
You are totally right. I forget that I can omit unused elements.
Maybe you can add a sample to rule description?

**Comment by vilchik-elena** [comment]
I've updated javascript:S1854
